python/tchannel/tornado/connection.py

```diff
@@ -44,51 +51,97 @@
 
 
 class TornadoConnection(object):
-    """Handle speaking TChannel over a Tornado connection."""
+    """Manages a bi-directional TChannel conversation between two machines.
+
+    The following primary bi-directional operations are provided:
+
+    ``write(message)``
+        Send the message up the wire.
+    ``await(message)``
+        Receive a message.
+    ``send(message)``
+        Send a message and receive its response.
+
+    In addition to those, the following operations are provided and should be
+    used depending on the direction of the connection.
+
+    ``initiate_handshake``
+        Perform a handshake with the remote host.
+    ``expect_handshake``
+        Expect a handshake request from the remote host.
+    """
+
+    # Message ID 0xffffffff is reserved
+    MAX_MESSAGE_ID = 0xfffffffe
+    CALL_TYPES = frozenset([Types.CALL_REQ, Types.CALL_REQ_CONTINUE])
 
     def __init__(self, connection):
-        self.connection = connection
+        assert connection
```

Comment:
Added a message.

Also, the trace will usually include the line at which the assertion failed so it's generally obvious for None checks why the assertion failed.